types/datasource.d.ts

```diff
@@ -84,6 +79,8 @@ export declare class DataSource extends EventEmitter {
 
   connector?: Connector;
 
+  definitions?: AnyObject;
```

Comment:
Please note `definitions` are always set to `modelBuilder.definitions`:

https://github.com/strongloop/loopback-datasource-juggler/blob/8d1690e9b2c76e3d0a8336b47f133e0605cbcc9a/lib/datasource.js#L139

We should use the same type for this property as we do in `ModelBuilder`:

https://github.com/strongloop/loopback-datasource-juggler/blob/8d1690e9b2c76e3d0a8336b47f133e0605cbcc9a/types/model.d.ts#L339